contracts/src/rollup/RollupCore.sol

```diff
@@ -502,7 +502,10 @@ abstract contract RollupCore is IRollupCore, PausableUpgradeable {
      */
     function deleteStaker(address stakerAddress) private {
         Staker storage staker = _stakerMap[stakerAddress];
-        require(staker.currentChallenge == NO_CHAL_INDEX, "STAKER_IN_CHALL");
+        require(
+            staker.currentChallenge == NO_CHAL_INDEX || isZombie(stakerAddress),
+            "STAKER_IN_CHALL"
+        );
```

Comment:
I thought we didn't want to allow zombies to re-stake?

Reply:
hmm so I guess we should not remove the zombie staker
https://github.com/OffchainLabs/nitro/blob/5e805ae5b6351f7db84e5e8eca37463caf600f5f/contracts/src/rollup/RollupCore.sol#L385

Reply:
if we don't remove the zombie from staker map, what is the expected behavior when the zombie is removed by `removeZombie`? Should we remove it from the staker map at that point? If not, they will not be able to stake on new node because `ALREADY_STAKED`

Reply:
Is this something we want https://github.com/OffchainLabs/nitro/commit/a1e03e802c2fa265fe42d2a29f15c71e26c9d19e?
will break a test case because a removed zombie trying to stake will hit ALREADY_STAKED